app/assets/v2/js/pages/bounty_detail/tezos_extension.js

```diff
@@ -0,0 +1,83 @@
+const payWithTezosExtension = async(fulfillment_id, to_address, vm, modal) => {
+
+  const amount = vm.fulfillment_context.amount;
+  const token_name = vm.bounty.token_name;
+
+  let selectedAddress;
+  
+  const Tezos = new taquito.TezosToolkit('https://mainnet-tezos.giganode.io');
+  const wallet = new taquitoBeaconWallet.BeaconWallet({ name: 'Gitcoin' });
+  
+  Tezos.setWalletProvider(wallet);
+
+  const activeAccount = await wallet.client.getActiveAccount();
+  
+  if (activeAccount) {
+    console.log('Already connected:', activeAccount.address);
+    selectedAddress = activeAccount.address;
+  } else {
+    try {
+      await wallet.requestPermissions();
+      selectedAddress = await wallet.getPKH();
+      console.log('New connection:', selectedAddress);
+    } catch (e) {
+      console.log(e);
+    }
+  }
+        
+  if (token_name == 'XTZ') {
```

Comment:
^ is there provision for non XTZ tokens ? If yes -> does the behavior change for those ?

Reply:
@thelostone-mc nope... there's no provision for none XTZ, should i take out the condition for now?

Reply:
Ah it's fine